### PR TITLE
fix: emit ValidationError instead of UnknownError in create telemetry

### PR DESCRIPTION
**Pull request by Hweinstock**
## Description

When the `create` command fails validation (e.g. missing `--framework`, `--model-provider`, `--memory`), telemetry emits `error_name: "UnknownError"` instead of `error_name: "ValidationError"`.

Two root causes:

1. **Plain `Error` thrown instead of `ValidationError`** — `command.tsx` throws `new Error(validation.error)`. The `classifyError` function only recognizes `BaseError` subclasses via `instanceof`, so a plain `Error` gets `{ category: "UnknownError", source: "unknown" }`.

2. **`minify: true` without `keepNames` mangles class names** — `BaseError` sets `this.name = this.constructor.name`. With minification, constructor names are mangled (e.g. `"Gr"`), so `ErrorName.safeParse(err.name)` fails for *all* error subclasses. Per [esbuild docs](https://esbuild.github.io/api/#keep-names):

   > However, minification renames symbols to reduce code size [...] That changes value of the `name` property for many of these cases. [...] some frameworks rely on the `name` property for registration and binding purposes. If this is the case, you can enable this option to preserve the original `name` values even in minified code.

### Fix

- `src/cli/commands/create/command.tsx` — throw `new ValidationError(...)` instead of `new Error(...)`
- `esbuild.config.mjs` — add `keepNames: true` to preserve class names through minification
- `integ-tests/create-edge-cases.test.ts` — assert `error_name: "ValidationError"` and `error_source: "user"`

## Related Issue

Closes #

## Type of Change

- [x] Bug fix

## Testing

- [x] I ran `npm run test:unit` and `npm run test:integ`
- [ ] I ran `npm run typecheck`
- [ ] I ran `npm run lint`
- [ ] If I modified `src/assets/`, I ran `npm run test:update-snapshots` and committed the updated snapshots

## Checklist

- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [ ] I have updated the documentation accordingly
- [x] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the
terms of your choice.

### CODE REVIEW

File: src/cli/primitives/GatewayTargetPrimitive.ts [hunk 1]
```diff
@@ -323,7 +324,7 @@ export class GatewayTargetPrimitive extends BasePrimitive<AddGatewayTargetOption
         await runCliCommand('add.gateway-target', !!cliOptions.json, async () => {
           const validation = await validateAddGatewayTargetOptions(cliOptions);
           if (!validation.valid) {
-            throw new Error(validation.error);
+            throw new ValidationError(validation.error!);
```
Comment (Hweinstock) [inline src/cli/primitives/GatewayTargetPrimitive.ts:327]: The `!` assertion is necessary because the error is optional on the type: https://github.com/aws/agentcore-cli/blob/2d8b9d7bfcf155cc846eb680a1ae96b9f9560ea9/src/cli/commands/add/validate.ts. Hope to refactor this to use common result shape as a follow-up.

File: src/cli/commands/create/command.tsx [hunk 2]
```diff
@@ -132,7 +132,7 @@ async function handleCreateCLI(options: CreateOptions): Promise<void> {
     async () => {
       const validation = validateCreateOptions(options, cwd);
       if (!validation.valid) {
-        throw new Error(validation.error);
+        throw new ValidationError(validation.error!);
```
[the remark this answers is not in the document]
Comment (Hweinstock) [reply src/cli/commands/create/command.tsx:135]: ooh thank you!

**Review by avi-alpert — APPROVED** [review]
